### ISSUE TITLE
"ReferenceError: _this is not defined" when capturing this in super call

### ISSUE THREAD

**Issue by tnikodym** [Microsoft/TypeScript#13058]
**TypeScript Version:**  2.1.4

**Code**

```ts
constructor() {
        super(null, (arg) => this.myMethod(arg));
    }
```

**Emitted code TypeScript@2.1.4:**

```js
    function MyClass() {
        // Note the use of _this, which is not defined 
        // Throws "ReferenceError: _this is not defined"
        return _super.call(this, null, function (arg) { return _this.myMethod(arg); }) || this;
    }
```

**Emitted code TypeScript@2.0.10:**

```js
    function MyClass() {
        var _this = this;
        _super.call(this, null, function (arg) { return _this.myMethod(arg); });
    }
```

**Actual behavior:**

No error is reported at compile time and the code throws ReferenceError in runtime. 

**Note:** 

The same TS code works fine when compiled in older versions of tsc (e.g. 2.0.10)

**Comment by mhegazy** [comment]
looks like a duplicate of https://github.com/Microsoft/TypeScript/issues/12764. Can you give `typescript@2.2.0-dev.20161215` or later a try?
